util-taglib/src/com/liferay/taglib/util/InlineUtil.java

```diff
@@ -33,10 +35,21 @@ public static String buildDynamicAttributes(
 
 		StringBundler sb = new StringBundler(dynamicAttributes.size() * 4);
 
+		List<String> attributesList = Arrays.asList(_HTML_BOOLEAN_ATTRIBUTES);
+
 		for (Map.Entry<String, Object> entry : dynamicAttributes.entrySet()) {
 			String key = entry.getKey();
 
 			if (!key.equals("class")) {
+				if (attributesList.contains(key)) {
```

Comment:
Speaking of the algorithms course, what data structure lends itself to `O(1)` membership checks? (I know, doing an `O(n)` scan of an array with 25 items is probably not that much of an issue, even in the presence of a decently sized bunch of dynamic attributes but I still kind of like using a data structure that suits the access pattern.)

And not being a Java expert, is there any way to avoid the repeated conversion from array before we enter the `for` loop?